example/autoencoder/mnist_sae.py

```diff
@@ -15,30 +15,74 @@
 # specific language governing permissions and limitations
 # under the License.
 
-# pylint: skip-file
 from __future__ import print_function
+import argparse
 import mxnet as mx
 import numpy as np
 import logging
 import data
 from autoencoder import AutoEncoderModel
 
+parser = argparse.ArgumentParser(description='Train an auto-encoder model for mnist dataset.')
+parser.add_argument('--print-every', type=int, default=1000,
+                    help='the interval of printing during training.')
+parser.add_argument('--batch-size', type=int, default=256,
+                    help='the batch size used for training.')
+parser.add_argument('--pretrain-iter', type=int, default=50000,
+                    help='the number of iterations for pretraining.')
+parser.add_argument('--finetune-iter', type=int, default=100000,
+                    help='the number of iterations for fine-tuning.')
+parser.add_argument('--visualize', action='store_true',
+                    help='whether to visualize the original image and the reconstructed one.')
+parser.add_argument('--num-units', type=str, default="784,500,500,2000,10",
+                    help='the number of hidden units for the layers of the encoder.' \
+                         'The decoder layers are created in the reverse order.')
+
+# set to INFO to see less information during training
+logging.basicConfig(level=logging.DEBUG)
+opt = parser.parse_args()
+logging.info(opt)
+print_every = opt.print_every
+batch_size = opt.batch_size
+pretrain_iter = opt.pretrain_iter
+finetune_iter = opt.finetune_iter
```

Comment:
`pretrain_num_iter` and `finetune_num_iter` are more clear names since reader won't mistake these for iterators.